packages/flutter/lib/src/services/asset_bundle.dart

```diff
@@ -97,15 +99,17 @@ class NetworkAssetBundle extends AssetBundle {
   @override
   Future<ByteData> load(String key) async {
     final http.Response response = await _httpClient.get(_urlFromKey(key));
-    if (response.statusCode == 200)
-      return null;
+    if (response.statusCode != 200)
+      throw new FlutterError('Unable to load asset: $key');
```

Comment:
This should probably include information from `response` as well (like the status code, maybe more? I guess we don't want to include the whole body, that might be absurdly verbose).

See <https://docs.flutter.io/flutter/foundation/FlutterError/message.html> for some style notes about these error messages.